fe/fe-core/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java

```diff
@@ -791,18 +796,22 @@ private void updateNumOfData(long numOfTotalRows, long numOfErrorRows, long unse
         this.jobStatistic.currentTotalRows += numOfTotalRows;
         this.jobStatistic.errorRowsAfterResumed = this.jobStatistic.currentErrorRows;
         if (this.jobStatistic.currentTotalRows > maxBatchRows * 10) {
-            if (this.jobStatistic.currentErrorRows > maxErrorNum) {
+            if (this.jobStatistic.currentErrorRows > maxErrorNum
+                    || (this.jobStatistic.currentErrorRows / this.jobStatistic.currentTotalRows) > maxFilterRatio) {
```

Comment:
take care of dividing by zero exception. check the divisor.

Reply:
Before executing the calculation, it is ensured that this. jobStatistics. currentTotalRows>maxBatchRows * 10, so there should be no exception of zero.